### DISCUSSION TITLE
Replace shortid with uuid

### DISCUSSION THREAD

**Discussion by michael-s-molina** [apache/superset#18525] [Ideas]
**Is your feature request related to a problem? Please describe.**
[shortid](https://github.com/dylang/shortid) is deprecated, because the architecture is unsafe, it does not generate cryptographically secure ids and we can't rely on it to make IDs which are impossible to guess.

**Describe the solution you'd like**
We should replace its occurrences with [uuid](https://github.com/uuidjs/uuid) to generate secure ids and also to benefit from performance gains of using a native database type.

**Describe alternatives you've considered**
[nanoid](https://github.com/ai/nanoid/) was considered but the benefits of using a native database type seems more appealing.

@rusackas @villebro @etr2460

**Comment by graceguo-supercat** [comment]
@michael-s-molina in which places Superset need secure ids? I know many places use shortid as unique UI component id, it doesn't matter you can guess it out or not.

**Comment by rumbin** [comment]
For reasons of data analytics of this data on non-indexed data warehouses, I'd like to suggest [ULID](https://github.com/ulid/spec).

Very similar to the features of UUIF, but lexicographically sortable in a meaningful way.
